lib/awful/titlebar.lua

```diff
@@ -782,6 +879,12 @@ function titlebar.widget.button(c, name, selector, action)
             if state ~= "" then
                 state = "_" .. state
             end
+            -- try select user defined tooltip texts according to state
+            local tooltip_text = beautiful["titlebar_tooltip_messages_" .. name .. "_" .. img]
+                or beautiful["titlebar_tooltip_messages_" .. name]
+                or default_tooltip_messages[name .. "_" .. img]
+                or default_tooltip_messages[name]
+                or name
```

Comment:
nit: Since there is no other usage of this variable, I think it can be moved to the inside of the `if titlebar.enable_tooltip` block.

Reply:
Maybe I don't understand now the description. Did you mean move whole business logic which is written in the `update()` function to the condition `if titlebar.enable_tooltip then`?

I already tested this before. But it cannot work, because it needs updated parts of `img` and has to be implemented inside `update()` function in the actual place.